src/kernel/src/hal/platform/hyperlight/peb/mod.rs

```diff
@@ -196,15 +165,27 @@ impl ProcessEnvironmentBlock {
     }
 
     fn print(&mut self, message: &str) -> Result<(), Error> {
-        self.print_with_magic_port(message)
+        self.debug_print(message)
         // self.print_with_host_fn(message)
     }
 
     #[allow(dead_code)]
-    fn print_with_magic_port(&mut self, message: &str) -> Result<(), Error> {
-        for byte in message.bytes() {
+    fn debug_print(&mut self, message: &str) -> Result<(), Error> {
+        let port = OutBAction::DebugPrint as u16;
+        let data = message.as_bytes();
+
+        for chunk in data.chunks(4) {
+            let val = match chunk {
+                [a, b, c, d] => u32::from_le_bytes([*a, *b, *c, *d]),
+                [a, b, c] => u32::from_le_bytes([*a, *b, *c, 0x7F]),
+                [a, b] => u32::from_le_bytes([*a, *b, 0x7F, 0x7F]),
+                [a] => u32::from_le_bytes([*a, 0x7F, 0x7F, 0x7F]),
```

Comment:
Why `0x7f` and not `0xff`?

Reply:
0x7F padding is not printable in HL. 0xFF will output �

Reply:
Could you add a comment about this? This design decision is not obvious. For instance, I assumed that Hyperlight would scan the u32 for characters that are out of the printable ASCII range to handle the conversion manually.